dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala

```diff
@@ -0,0 +1,275 @@
+    println(s"Making requests for ${if (matching) "matching" else "placing"} ...")
+
+    mkOrders(accounts, mkPairsAndDistribute(accounts, pairsFile), matching).take(requestsCount)
+  }
+
+  private def mkPlaces(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File]): List[Request] =
+    mkPlaces(accounts, requestsCount, pairsFile: Option[File], matching = false)
+
+  private def mkMatching(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File]): List[Request] =
+    mkPlaces(accounts, requestsCount, pairsFile: Option[File], matching = true)
+
+  private def mkCancels(accounts: List[PrivateKeyAccount], requestsCount: Int): List[Request] = {
+    println("Making requests for cancelling...")
+
+    val cancels = accounts
+      .flatMap(a => {
+        Json
+          .parse(getOrderBook(a))
+          .as[List[JsValue]]
+          .map(o => {
+            val id = (o \ "id").as[String]
+            val aa = ((o \ "assetPair").as[JsValue] \ "amountAsset").validate[String] match {
+              case JsSuccess(name, _) => name
+              case _: JsError         => "WAVES"
+            }
+            val pa = ((o \ "assetPair").as[JsValue] \ "priceAsset").validate[String] match {
+              case JsSuccess(name, _) => name
+              case _: JsError         => "WAVES"
+            }
+            Request(Request.POST.toString,
+                    s"/matcher/orderbook/$aa/$pa/cancel",
+                    Request.CANCEL.toString,
+                    Transactions.makeOrderCancel(a, new AssetPair(aa, pa), id))
+          })
+      })
+
+    cancels.take(requestsCount)
+  }
+
+  private def mkOrderHistory(accounts: List[PrivateKeyAccount], requestsCount: Int, pairsFile: Option[File]): List[Request] = {
+    println("Making requests for getting order history...")
+
+    val pairs = readAssetPairs(pairsFile)
+    val ts    = System.currentTimeMillis
+    val oha   = settings.distribution.getOrElse(Request.ORDER_HISTORY_BY_ACC.toString, 1.0)
+    val ohp   = settings.distribution.getOrElse(Request.ORDER_HISTORY_BY_PAIR.toString, 1.0)
+
+    val all = accounts
+      .flatMap(a => {
+        Request(
+          Request.GET.toString,
+          s"/matcher/orderbook/${Base58.encode(a.getPublicKey)}",
+          Request.ORDER_HISTORY_BY_ACC.toString,
+          headers = Map("Signature" -> services.matcher.getOrderHistorySignature(a, ts), "Timestamp" -> ts.toString)
+        ) ::
+          pairs.map(
+          p =>
+            Request(
+              Request.GET.toString,
+              s"/matcher/orderbook/${p.getAmountAsset}/${p.getPriceAsset}/publicKey/${Base58.encode(a.getPublicKey())}",
+              Request.ORDER_HISTORY_BY_PAIR.toString,
+              headers = Map("Signature" -> services.matcher.getOrderHistorySignature(a, ts), "Timestamp" -> ts.toString)
```

Comment:
Seems that request creation can be extracted in a separate function

Reply:
fixed